cropduster/models.py

```diff
@@ -522,7 +524,9 @@ def _save_standalone_thumb(self, size, image=None, thumb=None, commit=True):
         return thumb
 
     def _save_thumb(self, size, image=None, thumb=None, ref_thumb=None, tmp=False, commit=True):
-        image = image or PIL.Image.open(safe_str_path(self.image.path))
+        with self.image as f:
```

Comment:
In cases where the image is passed in as an argument, this would cause redundant file open operations. Better to have the `with self.image as f: f.open()` inside a conditional so that it only gets executed if an image isn't passed in, e.g.

```python
if not image:
    with self.image as f:
        f.open()
        image = PIL.Image.open(BytesIO(f.read()))
```